src/sgame/lua/Interpreter.cpp

```diff
@@ -0,0 +1,239 @@
+/*
+===========================================================================
+
+Unvanquished GPL Source Code
+Copyright (C) 2024 Unvanquished Developers
+
+This file is part of the Unvanquished GPL Source Code (Unvanquished Source Code).
+
+Unvanquished Source Code is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Unvanquished Source Code is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Unvanquished Source Code.  If not, see <http://www.gnu.org/licenses/>.
+
+In addition, the Unvanquished Source Code is also subject to certain additional terms.
+You should have received a copy of these additional terms immediately following the
+terms and conditions of the GNU General Public License which accompanied the Unvanquished
+Source Code.  If not, please request a copy in writing from id Software at the address
+below.
+
+If you have questions concerning this license or the applicable additional terms, you
+may contact in writing id Software LLC, c/o ZeniMax Media Inc., Suite 120, Rockville,
+Maryland 20850 USA.
+
+===========================================================================
+*/
+
+#include "sgame/lua/Interpreter.h"
+
+#include "common/Command.h"
+#include "common/FileSystem.h"
+#include "shared/bg_lua.h"
+#include "shared/lua/LuaLib.h"
+#include "shared/lua/Utils.h"
+#include "sgame/sg_local.h"
+
+using Shared::Lua::LuaLib;
+
+namespace Lua {
+
+static lua_State* L = nullptr;
+
+bool LoadCode( Str::StringRef code, Str::StringRef location )
+{
+	if ( luaL_loadbuffer( L, code.c_str(), code.size(), location.c_str() ) != 0 )
+	{
+		Shared::Lua::Report( L, "Loading buffer" );
+		return false;
+	}
+	return true;
+}
+
+bool RunCode()
+{
+	if ( lua_pcall( L, 0, 0, 0 ) != 0 )
+	{
+		Shared::Lua::Report( L, "Executing code" );
```

Comment:
This just prints out `Warn: Executing code` if there is an error in the script, which doesn't seem very helpful. Same goes for `Loading buffer`

Reply:
It also prints out the issue. While it is feasible to wrap this in a macro with __FILE__ and __LINE__, it seems a bit overkill?

It returns a bool and the call site should also print out a warning that something went wrong.

Reply:
Maybe I should say "Error executing Lua:" instead.

Reply:
That's fine

Reply:
Done.